x-pack/plugins/apm/server/lib/service_group_query_with_overflow.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { QueryDslQueryContainer } from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import { kqlQuery } from '@kbn/observability-plugin/server';
+import { SERVICE_NAME } from '../../common/es_fields/apm';
+import { ServiceGroup } from '../../common/service_groups';
+
+export function serviceGroupWithOverflowQuery(
+  serviceGroup?: ServiceGroup | null
+): QueryDslQueryContainer[] {
+  if (serviceGroup) {
+    const serviceGroupQuery = kqlQuery(serviceGroup?.kuery);
+    const otherBucketQuery = kqlQuery(`${SERVICE_NAME} : "_other"`);
+
+    return [
+      {
+        bool: {
+          should: [...serviceGroupQuery, ...otherBucketQuery],
```

Comment:
I think it's better to explicitly wrap both:

```
{
	bool: {
		should: [
			{
				bool: {
					filter: serviceGroupQuery
				}
			},
			{
				bool: {
					filter: [
						{
							term: {
								[SERVICE_NAME]: '_other'
							}
						}
					]
				}
			}	
		],
		minimum_should_match: 1
	}
}
```

This way you don't depend on the fact `kqlQuery` wraps its query in a `bool`.

Reply:
I don't understand. Basically the change i did will exactly output the same thing ?
Is it possible that the logic around kqlQuery wrapping its query in a bool can change ?